### PR TITLE
Improve readme

### PR DESCRIPTION
**Pull request by Marzz29**
**I have made changes to the README.md file in the repository, but I have not yet started working on the documentation for individual Cloud service providers. I suggest hosting the individual documentations on Gitbook.**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Changed
 
+- added documentation links for each provider to the readme.
+- added `doc.md` files to each cloud service provider's package folder to the readme.
+- added a comparison of the conventional method with Bifrost's methods for uploading files to GCS and Pinata to the readme.
```
Comment (opensaucerer) [inline changelog.md:7]: Hi @Marzz29 
Great work on this. 

For this part, however, I think these changes are new, they never existed before and should, probably be classified under the `## New` section.

It's thoughful that you kept it under the same version `v0.0.6`.

File: readme.md [hunk 2]
```diff
@@ -2,25 +2,81 @@
...
+- [Changelog](#changelog)
+- [Contributors](#contributors)
 
 # Problem Statement
+Many projects need to store files in the cloud, and different projects might use different cloud storage providers. Using different SDKs with different implementations for each provider can be tedious and time-consuming. Bifrost aims to simplify the process of working with multiple cloud storage providers by providing a consistent API for all of them. To better understand how Bifrost solves this problem, let's take a look at two separate code samples for GCS and Pinata using conventional means and how Bifrost eases the steps. 
 
-You might just want to use 3 different cloud storage providers in your project. This means you'll need 3 different SDKs with 3 different implementations. That's just too much learning curve.
 
-How about you ride with Thor on the Bifrost and easily transport your files to any cloud storage with the exact same function calls.
+## Google Cloud Storage(GCS)
```
Comment (opensaucerer) [inline readme.md:24]: For this, I think the presentation here doesn't appear as in the pattern of the problem statement. Bifrost code even looks longer...haha. 

The idea is that we should create two sample codes in sequence... 

## Google Cloud Storage using GCS SDK
```go
// insert code sample sing GCS SDK (same one you already have)
```

## Pinata Cloud using Pinata API
```go
// insert code sample using Pinata API (same one you already have)
```

Then we add a line saying something like
> From the sample above, it's easy to see that using both GCS and Pinata in your Go project will require you to learn to use two different tools with separate implementation patterns. Now, this is where Bifrost comes in. Mount bridges to the providers you want and call the exact same function when you need to upload files using any of these bridges. Now let's rewrite the above the Bifrost way. 

```go
package main

import (
	"fmt"
	"os"
	"github.com/opensaucerer/bifrost"
) 

// mount a bridge to gcs
gcsBridge, _ := bifrost.NewRainbowBridge(&bifrost.BridgeConfig{
	DefaultBucket:   "bifrost",
	DefaultTimeout:  10,
	Provider:        bifrost.GoogleCloudStorage,
	CredentialsFile: "/path/to/service/account/json", // this is not required if you are using google's default credentials
	EnableDebug:     true,
	PublicRead:      true,
})
defer gcsBridge.Disconnect()
fmt.Printf("Connected to %s\n", bridge.Config().Provider)

// mount a bridge to Pinata
pinataBridge, _ := bifrost.NewRainbowBridge(&bifrost.BridgeConfig{
	Provider:    bifrost.PinataCloud,
	PinataJWT:   os.Getenv("PINATA_JWT"),
	EnableDebug: true,
	PublicRead:  true,
})
defer pinataBridge.Disconnect()
fmt.Printf("Connected to %s\n", bridge.Config().Provider)

// upload a file to gcs using the bridge
guf, _ := gcsBridge.UploadFile(bifrost.File{
	Path:     "../shared/image/aand.png",
	Filename: "a_and_ampersand.png",
	Options: map[string]interface{}{
		bifrost.OptMetadata: map[string]string{
			"originalname": "aand.png",
		},
	},
})
fmt.Printf("Uploaded to GCS file: %s to %s\n", guf.Name, guf.Preview)

// upload a file to Pinata using the bridge
puf, _ := bridge.UploadFile(bifrost.File{
	Path:     "../shared/image/aand.png",
	Filename: "pinata_aand.png",
	Options: map[string]interface{}{
		bifrost.OptPinata: map[string]interface{}{
			"cidVersion": 1,
		},
		bifrost.OptMetadata: map[string]string{
			"originalname": "aand.png",
		},
	},
})
fmt.Printf("Uploaded to Pinata file: %s to %s\n", puf.Name, puf.Preview)
```

Then, after the above, we can add another closing line saying something like
> From the above example, it's easy to see the speed, simplicity, and ease of use that comes with Bifrost. Now you know what it feels like to ride with Thor!

File: readme.md [hunk 3]
```diff
@@ -2,25 +2,81 @@
...
+  - [Pinata](#pinata)
+- [Installation](#installation)
+- [Usage](#usage)
+- [Contributing](#contributing)
+- [License](#license)
+- [Changelog](#changelog)
+- [Contributors](#contributors)
 
 # Problem Statement
+Many projects need to store files in the cloud, and different projects might use different cloud storage providers. Using different SDKs with different implementations for each provider can be tedious and time-consuming. Bifrost aims to simplify the process of working with multiple cloud storage providers by providing a consistent API for all of them. To better understand how Bifrost solves this problem, let's take a look at two separate code samples for GCS and Pinata using conventional means and how Bifrost eases the steps.
```
Comment (opensaucerer) [inline readme.md:21]: Please don't remove the Thor reference...that's where the project idea came from. 
I think we can have this as 
```go
/*
Many projects need to store files in the cloud and different projects might use different cloud storage providers or, sometimes, multiple cloud providers all at once. Using different SDKs with different implementations for each provider can be tedious and time-consuming. Bifrost aims to simplify the process of working with multiple cloud storage providers by providing a consistent API for all of them. 

To better understand how Bifrost solves this problem, let's take you on a ride with Thor as we compare two separate code samples (conventional - bifrost) for using both Google Cloud Storage and Pinata Cloud in a single project.
*/
```

File: readme.md [hunk 4]
```diff
@@ -249,48 +245,16 @@ func main() {
...
+# Installation
+To install the Bifrost package, run the following command in your terminal:
+```bash
+go get github.com/opensaucerer/bifrost
 ```
+# Usage
+If you want to learn more about how Bifrost is creating different methods to make it easier to use different cloud providers, you can follow these links: 
+- [Google Cloud Storage (GCS)](gcs\doc.md)
+- [Amazon S3](s3\doc.md)
+- [Pinata](pinata\doc.md)
```
Comment (opensaucerer) [inline readme.md:257]: `Pinata Cloud`

**Review by opensaucerer — COMMENTED** [review]
Thank you @Marzz29 just some changes before we get this approved. 

And for the sub docs in each package, for now, let's just move the code samples present in the readme into each of them for each provider. 

Go packages, by convention, don't use GitBook for documentation. All documentation tends to remain on GitHub or appear on a separate website.

File: changelog.md (hunk 1 above)
Comment (Marzz29) [reply changelog.md:7]: Thank you for your response. I am currently working on implementing the changes.

File: readme.md (hunk 2 above)
Comment (Marzz29) [reply readme.md]: I've resolved the requested changes and it's ready for review.

File: readme.md (hunk 4 above)
Comment (Marzz29) [reply readme.md]: I've resolved the requested changes and it's ready for review.

**Review by opensaucerer — APPROVED** [review]
LGTM

**Comment by opensaucerer** [comment]
@Marzz29 Thank you for this contribution. Based on my last comment above and because we've removed the docs from the readme, it's important that we updated the `doc.md` of each provider as quick as we can to ensure all links work as expected.

**Comment by Marzz29** [comment]
Great, I'll begin working on the `docs.md` for each provider.

**Pull request merged**